src/ontodia/diagram/linkView.ts

```diff
@@ -60,14 +64,28 @@ export class LinkView extends joint.dia.LinkView {
         }
 
         const showLabels = typeModel && typeModel.get('showLabel');
-        const labelAttributes = showLabels ? [{
-            position: 0.5,
-            attrs: {text: {
-                text: this.view.getLinkLabel(linkTypeId).text,
-            }},
-        }] : [];
-
-        merge(linkAttributes, {labels: labelAttributes});
+        const labelAttributes: joint.dia.LinkLabelAttributes[] = [];
+
+        if (showLabels) {
+            const label = {...style.label};
```

Comment:
The shallow clone could cause an issue because of mutating this user-provided object later (in a `merge` call).